### PR TITLE
Group some NPM dependency updates

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Group some npm dependencies that should be upgraded together (e.g. all vite or monaco dependencies)

## Test Plan

I validate the configuration with `renovate-config-validator`

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+    "prEditedNotification"
+  ],
+  "extends": [
+    "config:recommended"
+  ],
+  "labels": [
+    "internal"
+  ],
+  "schedule": [
+    "before 4am on Monday"
```
Comment (AlexWaygood) [inline .github/renovate.json5:14]: nit: this is a `.json5` file, not `.json`, so we can use trailing commas to reduce diffs from future changes to this file:

```suggestion
  "suppressNotifications": [
    "prEditedNotification",
  ],
  "extends": [
    "config:recommended",
  ],
  "labels": [
    "internal",
  ],
  "schedule": [
    "before 4am on Monday",
```

Did prettier or another autoformatter make the changes to these lines?

File: .github/renovate.json5 [hunk 2]
```diff
@@ -1,51 +1,53 @@
...
-  "separateMajorMinor": false,
-  "enabledManagers": [
-    "github-actions",
-    "pre-commit",
-    "cargo",
-    "pep621",
-    "npm",
-  ],
-  "cargo": {
+  $schema: "https://docs.renovatebot.com/renovate-schema.json",
```
Comment (MichaReiser) [inline .github/renovate.json5:2]: I Formatted the file with Prettier, it removes the unnecessary quotes.

File: .github/renovate.json5 (hunk 1 above)
Comment (MichaReiser) [reply .github/renovate.json5]: It was jetbrains. i disabled the autoformatter for JSON and instead ran prettier (which is what we use for other files)

Comment (AlexWaygood) [reply .github/renovate.json5]: I like the prettier changes more than the JetBrains changes :) though it's a _little_ hard to see here which changes have an impact on the config and which are just cosmetic

File: .github/renovate.json5 [hunk 3]
```diff
@@ -1,51 +1,53 @@
...
+      description: "Weekly update of NPM development dependencies",
     },
     {
-      "groupName": "pre-commit dependencies",
-      "matchManagers": ["pre-commit"],
-      "description": "Weekly update of pre-commit dependencies",
+      groupName: "Monaco",
+      matchManagers: ["npm"],
+      matchPackagePatterns: ["monaco"],
+      description: "Weekly update of the Monaco editor",
```
Comment (AlexWaygood) [inline .github/renovate.json5:46]: This group currently just includes `@monaco-editor/react` and `monaco-editor`, right? (Just checking I understand correctly)

Comment (MichaReiser) [reply .github/renovate.json5:46]: Yes, the main reason is that it doesn't make sense to update them separately.

**Comment by AlexWaygood** [comment]
Thanks!

**Pull request merged**